src/citation/permissions.py

```diff
@@ -15,3 +16,13 @@ def has_permission(self, request, view):
     def has_object_permission(self, request, view, obj):
         requestor = request.user
         return obj.get_is_user_admin(requestor)
+
+
+class PDFUploadsS3CallBack(BasePermission):
+    message = "Permission Denied: Endpoint restricted to S3 Lambda trigger"
+
+    def has_permission(self, request, view):
+        client_ip = get_client_ip(request)
+        if client_ip not in LAMBDA_IPS:
```

Comment:
Good thinking, I imagine we were able to get a static list of ips?

Reply:
Yeah, I configured the Lambda triggers to be connected within a couple of subnets in our VPC. Each subnet is connected to NAT gateway and each gateway has an elastic ip addressed attached to it. There are 2 elastic addresses assigned, one for prod and the other for staging.